pkg/sql/bulksst/sst_file_allocator.go

```diff
@@ -101,7 +108,7 @@ func NewExternalFileAllocator(es cloud.ExternalStorage, baseURI string) FileAllo
 func (e *ExternalFileAllocator) AddFile(
 	ctx context.Context, fileIndex int, span roachpb.Span, rowSample roachpb.Key, fileSize uint64,
 ) (objstorage.Writable, func(), error) {
-	fileName := fmt.Sprintf("%d.sst", fileIndex)
+	fileName := fmt.Sprintf("%s_%d.sst", e.baseFileName, fileIndex)
```

Comment:
I have an alternative fix for this in my integration path. I decided to pass the whole base uri to the allocator. I like that approach because it mirrors how cloud.ExternalStorage prefix/basename pairs work.

Reply:
I'll close this then 😇